CogniteSdk.Types/Beta/StreamRecords/StreamRecordAggregate.cs

```diff
@@ -0,0 +1,349 @@
+// Copyright 2025 Cognite AS
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Generic;
+using System.Text.Json;
+using CogniteSdk.DataModels;
+
+namespace CogniteSdk.Beta
+{
+    /// <summary>
+    /// Request for aggregating records in a stream.
+    /// </summary>
+    public class StreamRecordsAggregate
+    {
+        /// <summary>
+        /// Filter on records created within the provided range.
+        /// </summary>
+        public LastUpdatedTimeFilter LastUpdatedTime { get; set; }
+
+        /// <summary>
+        /// A filter Domain Specific Language (DSL) used to create advanced filter queries.
+        /// </summary>
+        public IDMSFilter Filter { get; set; }
+
+        /// <summary>
+        /// A dictionary of requested aggregates with client defined names/identifiers.
+        /// </summary>
+        public Dictionary<string, StreamRecordAggregateDefinition> Aggregates { get; set; }
+    }
+
+    /// <summary>
+    /// Base class for aggregate definitions. Use one of the derived classes to specify the aggregate type.
+    /// </summary>
+    public class StreamRecordAggregateDefinition
+    {
+        /// <summary>
+        /// Average aggregate. Returns the average value of a numeric property.
+        /// </summary>
+        public StreamRecordPropertyAggregate Avg { get; set; }
```

Comment:
This is a larger change, but I think the proper way to model this is using the `ExtTaggedUnionConverter`. With that you'd define a marker interface,

`IStreamRecordAggregate`, and then an aggregate type for each `AvgAggregate`, `CountAggregate`, etc. then set `Suffix` to "Aggregate" in a type extending the `ExtTaggedUnionConverter`... Not trivial, but it would model this more correctly, I think.